src/Microsoft.Health.Fhir.SqlServer/Features/Search/HashingSqlQueryParameterManager.cs

```diff
@@ -0,0 +1,201 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Security.Cryptography;
+using EnsureThat;
+using Microsoft.Data.SqlClient;
+using Microsoft.Health.SqlServer;
+using Microsoft.Health.SqlServer.Features.Schema.Model;
+using Microsoft.Health.SqlServer.Features.Storage;
+
+namespace Microsoft.Health.Fhir.SqlServer.Features.Search
+{
+    /// <summary>
+    /// Wraps a <see cref="SqlQueryParameterManager"/>, adding the ability to compute a hash of a subset of the parameters.
+    /// </summary>
+    public class HashingSqlQueryParameterManager
+    {
+        private readonly SqlQueryParameterManager _inner;
+        private readonly HashSet<SqlParameter> _setToHash = new();
+
+        public HashingSqlQueryParameterManager(SqlQueryParameterManager inner)
+        {
+            EnsureArg.IsNotNull(inner, nameof(inner));
+            _inner = inner;
+        }
+
+        public bool HasParametersToHash => _setToHash.Count > 0;
+
+        public SqlParameter AddParameter<T>(Column<T> column, T value, bool includeInHash = true)
+        {
+            return AddParameter((Column)column, value, includeInHash);
+        }
+
+        public SqlParameter AddParameter(Column column, object value, bool includeInHash = true)
```

Comment:
>bool includeInHash = true) [](start = 70, length = 26)

nit: would it be better to make this a non-optional parameter so that callers explicitly set `includeInHash`. I feel that having a default value in this situation makes it easy for someone in the future to introduce a bug by calling `AddParameter()` without realizing what the default behavior does.

Reply:
Good idea. Updated.